addons/udes_stock/views/stock_picking.xml

```diff
@@ -127,6 +127,11 @@
                 <attribute name="invisible">1</attribute>
             </xpath>
 
+            <!--Add Set To Draft button visible when picking has been marked as to do-->
+            <xpath expr="//button[@name='action_confirm']" position="before">
+                <button name="button_draft" attrs="{'invisible': [('state', 'not in', ['waiting', 'confirmed'])]}" string="Set To Draft" type="object" class="oe_highlight" groups="base.group_user"/>
```

Comment:
Nit: `groups` should be `stock.group_stock_user`

Reply:
Nevermind, didn't realise other buttons were also using `base.group_user` (including unreserve), only a few use the stock group... I guess the logic is that if you have access to the pickings then you can press the buttons. :)

Reply:
Yep, I only checked after adding the comment. :sweat_smile: 

Although with the example you gave there is another super super nit... `Set to Draft` to match with `Mark as Todo`? (to isn't capitalised in a title) - happy for you to leave that though :smile: